compiler/rustc_lint/src/dangling.rs

```diff
@@ -133,10 +133,11 @@ impl DanglingPointerSearcher<'_, '_> {
 
 fn lint_expr(cx: &LateContext<'_>, expr: &Expr<'_>) {
     if let ExprKind::MethodCall(method, receiver, _args, _span) = expr.kind
-        && matches!(method.ident.name, sym::as_ptr | sym::as_mut_ptr)
+        && let Some(fn_id) = cx.typeck_results().type_dependent_def_id(method.hir_id)
```

Comment:
The hir id of `method` won't resolve to the function, for that you should use the one from the method call it-self, in this case from `expr`.
```suggestion
        && let Some(fn_id) = cx.typeck_results().type_dependent_def_id(expr.hir_id)
```